app/models/good_job/process.rb

```diff
@@ -12,6 +12,22 @@ class Process < BaseRecord
     STALE_INTERVAL = 30.seconds
     # Interval until the process record is treated as expired
     EXPIRED_INTERVAL = 5.minutes
+    PROCESS_MEMORY = case RUBY_PLATFORM
+                     when /linux/
+                       ->(pid) {
+                         File.readlines("/proc/#{pid}/status").each do |line|
```

Comment:
Can you switch this over to `/proc/#{pid}/smaps_rollup`? The older status file doesn't contain PSS which will be more accurate in certain scenarios: https://www.baeldung.com/linux/process-memory-management#pss-memory. It would be nice if you could check if MacOS `ps` also supports pss output.

Sidekiq doesn't have to worry about such things I believe (except maybe in enterprise which supports forking?). It has already been available for almost a decade, but to be safe lets wrap it in rescue ENOENT. The code should be the same except that you're checking for `Pss:`

Reply:
Fixed. Macos doesn't have `proc` concept while `man ps` doesn't have any Pss mentioned there.